			     BASH PATCH REPORT
			     =================

Bash-Release:	4.0
Patch-ID:	bash40-011

Bug-Reported-by:	Matt Zyzik <Matt@ice.filescope.com>n
Bug-Reference-ID:	<20090312015018.C00741383ED@ice.filescope.com>
Bug-Reference-URL:	http://lists.gnu.org/archive/html/bug-bash/2009-03/msg00092.html

Bug-Description:

When using the new |& operator following a simple command with a redirection,
the redirection of stderr through the pipe was not performed under certain
circumstances.

Patch:

*** bash-4.0-patched/parse.y	2009-03-08 21:24:47.000000000 -0400
--- bash-4.0-patched/parse.y	2009-03-12 21:36:23.000000000 -0400
***************
*** 1123,1127 ****
  			  REDIRECT *r;
  
! 			  tc = $1;
  			  rd.dest = 1;
  			  r = make_redirection (2, r_duplicating_output, rd);
--- 1123,1127 ----
  			  REDIRECT *r;
  
! 			  tc = $1->type == cm_simple ? (COMMAND *)$1->value.Simple : $1;
  			  rd.dest = 1;
  			  r = make_redirection (2, r_duplicating_output, rd);
*** bash-4.0/patchlevel.h	2009-01-04 14:32:40.000000000 -0500
--- bash-4.0-patched/patchlevel.h	2009-02-22 16:11:31.000000000 -0500
***************
*** 26,30 ****
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 10
  
  #endif /* _PATCHLEVEL_H_ */
--- 26,30 ----
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 11
  
  #endif /* _PATCHLEVEL_H_ */

